ray-curator/ray_curator/backends/experimental/ray_data/adapter.py

```diff
@@ -80,53 +72,81 @@ def process_dataset(self, dataset: Dataset) -> Dataset:
             msg = "Ray Data does not support nvdecs / nvencs. Please use gpus instead."
             raise ValueError(msg)
 
-        processed_dataset = dataset.map_batches(
-            create_named_ray_data_stage_adapter(self.stage).map_batch_fn,
-            batch_size=self.batch_size,
-            num_cpus=self.stage.resources.cpus,
-            num_gpus=self.stage.resources.gpus,
-        )
+        is_actor_stage_ = self.stage.ray_stage_spec().get(RayStageSpecKeys.IS_ACTOR_STAGE, is_actor_stage(self.stage))
```

Comment:
Nit: I like the concept but find it a bit verbose/difficult to follow. I wonder if it's easy enough for cases like these to have something like `self.stage.ray_stage_spec().is_actor_stage()` or if that leads to too much complexity.